Tests/ML/augmentations/test_transform_pipeline.py

```diff
@@ -0,0 +1,164 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import random
+
+import PIL
+import pytest
+import torch
+from torchvision.transforms import CenterCrop, ColorJitter, RandomAffine, RandomErasing, RandomHorizontalFlip, \
+    RandomResizedCrop, Resize, ToTensor
+from torchvision.transforms.functional import to_tensor
+
+from InnerEye.Common.common_util import is_windows
+from InnerEye.ML.augmentations.image_transforms import AddGaussianNoise, ElasticTransform, ExpandChannels, RandomGamma
+from InnerEye.ML.augmentations.transform_pipeline import ImageTransformationPipeline, \
+    create_cxr_transforms_from_config
+
+# create_transform_pipeline_from_config
+from Tests.SSL.test_data_modules import cxr_augmentation_config
+
+import numpy as np
+
+
+@pytest.mark.parametrize("use_different_transformation_per_channel", [True, False])
+def test_torchvision_on_various_input(use_different_transformation_per_channel: bool) -> None:
+    """
+    This tests that we can run transformation pipeline with out of the box torchvision transforms on various types
+    of input: PIL image, 3D tensor, 4D tensors. Tests that use_different_transformation_per_channel has the correct
+    behavior.
+    """
+    image = np.ones([256, 256]) * 255.
+    image[100:150, 100:200] = 1
+    transform = ImageTransformationPipeline(
+        [CenterCrop(224),
+         RandomErasing(),
+         RandomAffine(degrees=(10, 12), shear=15, translate=(0.1, 0.3))
+         ],
+        use_different_transformation_per_channel)
+
+    # Test PIL image input
+    image = PIL.Image.fromarray(image).convert("L")
+    transform(image)
+
+    # Test image as [C, H. W] tensor
+    image_as_tensor = to_tensor(image)
+    transform(image_as_tensor)
```

Comment:
maybe do some very basic asserts on the output? Shape? Type?

Reply:
updated